torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -272,20 +272,26 @@ void CudaPrinter::visit(const Load* v) {
       os() << "__half2float(" << *v->base_handle() << "[" << *v->flat_index()
            << "])";
     }
-  } else {
-    // Detects whether the load target is also a store target.
-    // TODO: this is currently too wide. It detects whether a store-target
-    // exists within the program. In fact, this check is only necessary within a
-    // kernel.
-    if (v->indices().empty()) {
-      os() << *v->base_handle();
-    } else if (!cuda_analysis_->is_buf_store_target(v->buf())) {
-      // Cuda __ldg can only be applied on read-only buffers.
-      os() << "__ldg(" << *v->base_handle() << " + " << *v->flat_index() << ")";
-    } else {
-      os() << *v->base_handle() << "[" << *v->flat_index() << "]";
-    }
+    return;
+  }
+  // Detects whether the load target is also a store target.
+  // TODO: this is currently too wide. It detects whether a store-target
+  // exists within the program. In fact, this check is only necessary within a
+  // kernel.
+  if (v->indices().empty()) {
+    os() << *v->base_handle();
+    return;
+  }
+  if (v->dtype().scalar_type() == ScalarType::Bool) {
```

Comment:
Why do we need to special case Bool here? A comment could be helpful.

Reply:
Good idea, I'll add it.  For posterity, there's no __ldg overload for bool, so we need to use normal array indexing.